### DISCUSSION TITLE
Writing a custom loader script in typescript

### DISCUSSION THREAD

**Discussion by jeremy-rifkin** [webpack/webpack#16713] [Q&A]
Hello,
I'm working on a project that has a custom loader script for pug files, `parsed_pug_file.js`, and I was looking at converting it to ts.

```js
            {
                test: /\.pug$/,
                loader: './etc/scripts/parsed_pug_file.ts',
                options: {
                    useGit: hasGit,
                },
            },
```

This works fine in development when our app is using `webpack-dev-middleware` but running from the command line with `npx webpack` fails with errors that appear to indicate the ts file is being parsed as js.

Is it possible to get this working? Why is webpack-dev-middleware implicitly handling the ts while the command line is not?

**Comment by alexander-akait** [comment]
You should not use `ts` files in the `loader` option, please convert `ts` to `js` firstly and use only `js` files

**Reply by jeremy-rifkin** [reply]
What allows the ts `loader` to work under webpack-dev-middleware?

**Reply by alexander-akait** [reply]
What do you mean? Please clarify, you just got `ts` file, it doesn't mean it works
